test/integration/cookbooks/docker-tests/recipes/default.rb

```diff
@@ -0,0 +1,25 @@
+package 'apparmor'
+package 'build-essential'
+
+include_recipe 'docker'
+ENV['CHEF_DRIVER'] = 'docker'
+
+bash 'build and install chef-provisioning-docker' do
+  cwd '/opt/chef-provisioning-docker'
+  code <<-EOS
+    rm chef-provisioning-docker-*.gem
+    /opt/chef/embedded/bin/gem build chef-provisioning-docker.gemspec
+  EOS
+  action :nothing
+end.run_action(:run)
+
+$:.unshift('/opt/chef-provisioning-docker/lib')
+require 'chef/provisioning/docker_driver/version'
+gem_dir = '/opt/chef-provisioning-docker'
+gem_path = File.join(gem_dir, "chef-provisioning-docker-#{Chef::Provisioning::DockerDriver::VERSION}.gem")
+gem_package 'chef-provisioning-docker' do
```

Comment:
I tried that and had issues getting it to load from a path. Seemed like that should have worked but even if it does, I don't believe it can install from raw source (a .gemspec and not .gem)